client/login/controller.js

```diff
@@ -16,6 +17,25 @@ const enhanceContextWithLogin = ( context ) => {
 		query,
 	} = context;
 
+	// Process a social login handoff from /start/user.
+	if ( query?.email_address && query?.source && query?.access_token && query?.id_token ) {
+		context.store.dispatch( {
+			type: SOCIAL_HANDOFF_CONNECT_ACCOUNT,
+			email: query.email_address,
+			authInfo: {
+				source: query.source,
+				access_token: query.access_token,
+				id_token: query.id_token,
+			},
+		} );
+		const params = new URLSearchParams( new URL( window.location.href ).search );
+		// Remove state-related data from URL but leave 'email_address'.
+		params.delete( 'source' );
+		params.delete( 'access_token' );
+		params.delete( 'id_token' );
+		page.redirect( window.location.pathname + '?' + params.toString() );
```

Comment:
Any ideas why `page.redirect()` adds a new history entry, instead of replacing the existing?

Reply:
I think it might be because it's running on the server-side, so `page.js` is doing the real redirect. I was getting server-side errors because it was trying to interact with `window.location` during server-side routing. I've made a change so that it only removes the query params on the client (it's sort of like an optional tidy-up I guess) and it's not adding an extra history entry anymore.